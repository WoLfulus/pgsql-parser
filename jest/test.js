const parser = require('../src');

const dosql = `
  DO $$
  BEGIN
      IF NOT EXISTS (
              SELECT
                  1
              FROM
                  pg_roles
              WHERE
                  rolname = 'administrator') THEN
              CREATE ROLE administrator;
              COMMENT ON ROLE administrator IS 'Administration group';
      END IF;
  END $$;
`;

describe('first', () => {
  it('parse and deparse', () => {
    const tree = parser.parse(dosql);
    expect(tree).toMatchSnapshot();
    const sql = parser.deparse(tree.query);
    expect(sql.trim()).toMatchSnapshot();
    expect(sql.trim()).toEqual(dosql.trim());
  });
});
