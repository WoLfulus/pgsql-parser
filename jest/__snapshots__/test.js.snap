// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first parse and deparse 1`] = `
Object {
  "query": Array [
    Object {
      "DoStmt": Object {
        "args": Array [
          Object {
            "DefElem": Object {
              "arg": Object {
                "String": Object {
                  "str": "
  BEGIN
      IF NOT EXISTS (
              SELECT
                  1
              FROM
                  pg_roles
              WHERE
                  rolname = 'administrator') THEN
              CREATE ROLE administrator;
              COMMENT ON ROLE administrator IS 'Administration group';
      END IF;
  END ",
                },
              },
              "defaction": 0,
              "defname": "as",
            },
          },
        ],
      },
    },
  ],
  "stderr": "",
}
`;

exports[`first parse and deparse 2`] = `
"DO $$
  BEGIN
      IF NOT EXISTS (
              SELECT
                  1
              FROM
                  pg_roles
              WHERE
                  rolname = 'administrator') THEN
              CREATE ROLE administrator;
              COMMENT ON ROLE administrator IS 'Administration group';
      END IF;
  END $$;"
`;
